src/Analysis/Ast/Impl/Diagnostics/DiagnosticsEntry.cs

```diff
@@ -51,6 +53,26 @@ public sealed class DiagnosticsEntry {
         /// </summary>
         public DiagnosticSource Source { get; }
 
+        public bool ShouldReport(IPythonModule module) {
+            // module should always be a user written python module
+            if (module.ModuleType != ModuleType.User || !(module is PythonModule pythonModule)) {
+                return false;
+            }
+
+            string line = pythonModule.GetLine(SourceSpan.Start.Line);
```

Comment:
You don't need to get a string line here. You can get the location of the comment using method similar to `AstExtensions.IsInsideComment`, something like `AstExtensions.HasComment(this PythonAst ast, string comment, int lineNumber)`.

Reply:
I think I see what you are saying, I've made some changes, can you tell me if I'm right?